core/src/com/fourinachamber/fortyfive/map/shop/PersonWidget.kt

```diff
@@ -0,0 +1,70 @@
+package com.fourinachamber.fortyfive.map.shop
+
+import com.badlogic.gdx.graphics.g2d.Batch
+import com.badlogic.gdx.scenes.scene2d.ui.Widget
+import com.badlogic.gdx.scenes.scene2d.utils.DragAndDrop
+import com.badlogic.gdx.scenes.scene2d.utils.Drawable
+import com.fourinachamber.fortyfive.screen.ResourceBorrower
+import com.fourinachamber.fortyfive.screen.ResourceManager
+import com.fourinachamber.fortyfive.screen.general.*
+import com.fourinachamber.fortyfive.screen.general.styles.StyleManager
+import com.fourinachamber.fortyfive.screen.general.styles.StyledActor
+import com.fourinachamber.fortyfive.screen.general.styles.addActorStyles
+import onj.value.OnjObject
+import com.fourinachamber.fortyfive.utils.plus
+import onj.value.OnjNamedObject
+
+class PersonWidget(
+    private val offsetX: Float,
+    private val offsetY: Float,
+    val scale: Float,
+    private val dropBehaviour: OnjNamedObject,
+    val screen: OnjScreen,
+) : Widget(), ResourceBorrower, StyledActor {
+
+    private lateinit var personDrawable: Drawable
+
+    private val defOffset: HashMap<String, Float> = HashMap()
+
+    init {
+        bindHoverStateListeners(this)
+    }
+
+    override fun draw(batch: Batch?, parentAlpha: Float) {
+        if (this::personDrawable.isInitialized) {
+            val xPos = offsetX + defOffset["offsetX"]
+            val yPos = offsetY + defOffset["offsetY"]
+            val w = personDrawable.minWidth * scale * defOffset["scale"]!!
+            val h = personDrawable.minHeight * scale * defOffset["scale"]!!
+            personDrawable.draw(batch, xPos, yPos, w, h)
+        }
+    }
+
+    fun setDrawable(imgData: OnjObject) {
+        val textureName = imgData.get<String>("textureName")
+        defOffset["offsetX"] = imgData.get<Double>("offsetX").toFloat()
+        defOffset["offsetY"] = imgData.get<Double>("offsetY").toFloat()
+        defOffset["scale"] = imgData.get<Double>("scale").toFloat()
+        ResourceManager.borrow(this, textureName)
+        personDrawable = ResourceManager.get(this, textureName)
+        ResourceManager.giveBack(this, textureName)
```

Comment:
the resource is not guaranteed to stay vaild after ResourceManager.giveBack is called, only call giveBack when you are done using the resource